src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrGetAliasMembershipResponse.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * * Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ */
+package com.rapid7.client.dcerpc.mssamr.messages;
+
+import java.io.IOException;
+import java.util.List;
+import com.rapid7.client.dcerpc.io.PacketInput;
+import com.rapid7.client.dcerpc.messages.RequestResponse;
+import com.rapid7.client.dcerpc.objects.RPCConformantIntegerArray;
+
+public class SamrGetAliasMembershipResponse extends RequestResponse {
+    private RPCConformantIntegerArray array;
+
+    public List<Integer> getList() {
+        if (array == null)
+            return null;
+        return array.getArray();
```

Comment:
Why does this return `List<Integer>` when you are passing in `new Integer[count]`?

Reply:
@tosmun-r7 We've been consistently providing List<> for the service layer so I didn't want to make the change here.

Reply:
Also we are wrapping it in unmodifiableList.

Reply:
What I've seen is that it's a mix. But this isn't even the service layer here. My point is this should just return the original array: https://github.com/rapid7/smbj-rpc/blob/e77b12f5a3e00c80e2501513ecc1d019c61709bc/src/main/java/com/rapid7/client/dcerpc/objects/RPCConformantArray.java#L45

Reply:
We shouldn't allow the content of a marshalled response to be mutable. If we simply return the T[] then it becomes mutable.